src/mame/atari/xybots.cpp

```diff
@@ -243,7 +242,9 @@ void xybots_state::video_int_ack_w(uint16_t data)
 uint16_t xybots_state::special_port1_r()
 {
 	int result = m_ffe200->read();
-	result ^= m_h256 ^= 0x0400;
+	result ^= m_h256;
+	if (!machine().side_effects_disabled())
+		m_h256 ^= 0x0400;
 	return result;
```

Comment:
Isn’t this going to change behaviour?  Think about associativity – assignment operators are right-associative in C++ (unlike other binary operators which are left-associative).  This:
```c++
	result ^= m_h256 ^= 0x0400;
```
Will be parsed as:
```c++
	result ^= (m_h256 ^= 0x0400);
```
Equivalent to:
```c++
	m_h256 ^= 0x0400;
	result ^= m_h256;
```

To do it with the side effects check, you’ll need to do:
```c++
	result ^= m_h256 ^ 0x0400;
	if (!machine().side_effects_disabled())
		m_h256 ^= 0x0400;
```